include/lcp-category.php

```diff
@@ -19,43 +19,22 @@ public static function get_instance(){
    * When the category is set using the `name` parameter.
    */
   public function with_name($name){
-    $lcp_category_id = null;
-    // AND relationship
-    if ( preg_match('/\+/', $name) ){
-      $categories = array();
-      $cat_array = explode("+", $name);
-
-      foreach ($cat_array as $category){
-        $categories[] = $this->get_category_id_by_name($category);
-      }
-      $lcp_category_id = $categories;
-      // OR relationship
-    } elseif (preg_match('/,/', $name )){
-      $categories = '';
-      $cat_array = explode(",", $name);
-
-      foreach ($cat_array as $category){
-        $id = $this->get_category_id_by_name($category);
-        $categories .= $id . ",";
-      }
-
-      $lcp_category_id = $categories;
-    } else {
-      $lcp_category_id = $this->get_category_id_by_name($name);
+    if ( preg_match('/\+/', $name) ){ // AND relationship
+      return $this->and_relationship($name);
+    } elseif (preg_match('/,/', $name )){ // OR relationship
+      return $this->or_relationship($name);
     }
-    return $lcp_category_id;
+    return $this->get_category_id_by_name($name);
   }
 
-  public function with_id($id){
-    if (preg_match('/\+/', $id)){
-      if ( preg_match('/(-[0-9]+)+/', $id, $matches) ){
+  public function with_id($cat_id){
+    if (preg_match('/\+/', $cat_id)){
+      if ( preg_match('/(-[0-9]+)+/', $cat_id, $matches) ){
         $this->exclude = implode(',', explode("-", ltrim($matches[0], '-') ));
       }
-      $lcp_category_id = array_map( 'intval', explode( "+", $id ) );
-    } else {
-      $lcp_category_id = $id;
+      return array_map( 'intval', explode( "+", $cat_id ) );
     }
-    return $lcp_category_id;
+    return $cat_id;
```

Comment:
This whole method is bugged, doing `id="1+2-3" doesn't work` (nothing to do with your refactoring, it's been so for a long time). I'm preparing a PR to merge with this branch so that category things should be good and done.